tests/foreman/cli/test_discoveryrule.py

```diff
@@ -242,143 +317,204 @@ def test_positive_update_org_loc(self):
+        rule = DiscoveryRule.info({'id': rule['id']})
+        self.assertEqual(rule['enabled'], 'true')
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_negative_update_name(self):
         """Update discovery rule name using invalid names only
 
         @id: 8293cc6a-d983-460a-b76e-221ad02b54b7
 
         @Assert: Rule name is not updated
-
-        @caseautomation: notautomated
         """
+        rule = self._make_discoveryrule()
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(CLIReturnCodeError):
+                    DiscoveryRule.update({'id': rule['id'], 'name': name})
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_negative_update_hostname(self):
         """Update discovery rule host name using number as a value
 
         @id: c382dbc7-9509-4060-9038-1617f7fef038
 
         @Assert: Rule host name is not updated
-
-        @caseautomation: notautomated
         """
+        rule = self._make_discoveryrule()
+        with self.assertRaises(CLIReturnCodeError):
+            DiscoveryRule.update({'id': rule['id'], 'hostname': '$#@!*'})
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_negative_update_limit(self):
         """Update discovery rule host limit using invalid values
 
         @id: e3257d8a-91b9-406f-bd74-0fd1fb05bb77
 
         @Assert: Rule host limit is not updated
-
-        @caseautomation: notautomated
         """
+        rule = self._make_discoveryrule()
+        host_limit = gen_string('alpha')
+        with self.assertRaises(CLIReturnCodeError):
+            DiscoveryRule.update({'id': rule['id'], 'hosts-limit': host_limit})
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_negative_update_priority(self):
         """Update discovery rule priority using invalid values
 
         @id: 0778dd00-aa19-4062-bdf3-752e1b546ec2
 
         @Assert: Rule priority is not updated
-
-        @caseautomation: notautomated
         """
+        rule = self._make_discoveryrule()
+        priority = gen_string('alpha')
+        with self.assertRaises(CLIReturnCodeError):
+            DiscoveryRule.update({'id': rule['id'], 'priority': priority})
+
+
+class DiscoveryRuleRoleTestCase(CLITestCase):
+    """Implements Foreman discovery Rules tests along with roles from CLI."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Tests for discovery rules via Hammer CLI"""
+        super(DiscoveryRuleRoleTestCase, cls).setUpClass()
+        cls.org = make_org()
+        cls.loc = make_location()
+        cls.hostgroup = make_hostgroup({
+            u'organization-ids': cls.org['id'],
+            u'location-ids': cls.loc['id'],
+        })
+        cls.password = gen_alphanumeric()
+        cls.user = make_user({
+            'organization-ids': cls.org['id'],
+            'password': cls.password,
+        })
+        cls.user['password'] = cls.password
```

Comment:
we explicitly did the assignment because user info doesn't have password key and if we read the password like `self.user['password'] `without above assignment then test may fail.